src/common/snippets/include/snippets/utils/params_dumper.hpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#ifdef SNIPPETS_DEBUG_CAPS
+
+#include "openvino/util/file_util.hpp"
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/pass/serialize_control_flow.hpp"
+#include "snippets/lowered/pass/serialize_data_flow.hpp"
+#include "snippets/utils/debug_caps_config.hpp"
+
+namespace ov {
+namespace snippets {
+
+class ParamsDump {
+public:
+    explicit ParamsDump(const ov::snippets::lowered::ExpressionPtr& expr,
+                        const DebugCapsConfig& debug_config,
+                        std::string params)
+        : expr(expr),
+          debug_config(debug_config),
+          params(std::move(params)) {
+        dump("_in");
+    }
+    ~ParamsDump() {
+        dump("_out");
+    }
+
+private:
+    void dump(const std::string&& postfix) const {
+        ov::util::create_directory_recursive(debug_config.dumpParams.dir);
+
+        if (const auto brgemm = ov::as_type_ptr<snippets::op::Brgemm>(expr->get_node())) {
+            std::string csv_path = debug_config.dumpParams.dir + "/brgemm.csv";
+            std::ofstream csv_file(csv_path, std::ios_base::app);
+            OPENVINO_ASSERT(csv_file.is_open(), "Can't open file for dumping brgemm params");
+            static int brgemm_num = 0;
+            if (brgemm_num == 0) {
+                csv_file << "name,in_type,out_type,in_shapes,out_shapes,M,N,K,m_block,n_block,k_block\n";
+            }
+            csv_file << brgemm->get_friendly_name() << "," << params << "\n";
```

Comment:
I think we shouldn't separate csv header printing from `params`. Can we move it to `params` collection logic?

Reply:
Moved